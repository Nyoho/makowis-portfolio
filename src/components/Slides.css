:local .slide p {
  display: none;
}