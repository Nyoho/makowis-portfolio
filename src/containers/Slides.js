// @flow
import { connect } from 'react-redux';

import Component from '../components/Slides';
import type { State } from '../reducers/index';

const mapStateToProps = ({ slides }: State) => ({ slides });

const Slides = connect(mapStateToProps)(Component);

export default Slides;
