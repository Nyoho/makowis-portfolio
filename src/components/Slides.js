// @flow
import React, { Children } from 'react';
import type { Node } from 'react';
import type { Slide } from '../models/Slide';
import { slide as className } from './Slides.css';

const SlideComponent = ({ id, content, link, title }: Slide) => (
  <a key={id} href={link.href} className="col mb-2">
    <span
      // eslint-disable-next-line react/no-danger
      dangerouslySetInnerHTML={{ __html: content.content }}
      className={className}
    />
    <p className="text-white my-1">{title}</p>
  </a>
);

const Wrapper = ({ children }: { children: Node }) => {
  if (children === null) {
    return null;
  }
  if (Children.count(children) <= 0) {
    return null;
  }
  return <div className="row my-3">{children}</div>;
};

type Props = {
  slides: Slide[],
};

const Slides = ({ slides }: Props) => (
  <section className="text-center text-white bg-warning p-5">
    <h2>Slides</h2>
    <Wrapper>{slides.map(SlideComponent)}</Wrapper>
  </section>
);

export default Slides;
