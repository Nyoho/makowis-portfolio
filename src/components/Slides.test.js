/* eslint-env jest */
// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Slides from './Slides';

it('renders Person', () => {
  const tree = renderer.create(<Slides slides={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
