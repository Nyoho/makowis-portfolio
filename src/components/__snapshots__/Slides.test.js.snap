// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Person 1`] = `
<section
  className="text-center text-white bg-warning p-5"
>
  <h2>
    Slides
  </h2>
  <div
    className="row my-3"
  />
</section>
`;
